src/core/bsequence.ml

```diff
@@ -0,0 +1,49 @@
+
+type +'a node =
+    | Nil
+    | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
```

Comment:
Well, `return` reflects that a sequence is monadic, but I guess it is okay to rename it something else.